java/core/src/java/org/apache/orc/filter/impl/OrFilter.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.orc.filter.impl;
+
+import org.apache.orc.OrcFilterContext;
+import org.apache.orc.filter.VectorFilter;
+
+public class OrFilter implements VectorFilter {
+
+  public final VectorFilter[] filters;
+  private final Selected orOut = new Selected();
+
+  public OrFilter(VectorFilter[] filters) {
+    this.filters = filters;
+  }
+
+  public static void merge(Selected src, Selected tgt) {
```

Comment:
This should be moved to Selected and renamed to unionDistinct.

Reply:
moved and named it as unionDisjoint